apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1099,32 +1118,25 @@ function validateAssessment({
   };
   assessment.zones.forEach((zone) => {
     zone.questions.map((zoneQuestion) => {
-      const autoPoints = zoneQuestion.autoPoints ?? zoneQuestion.points;
-      if (!allowRealTimeGrading && Array.isArray(autoPoints) && autoPoints.length > 1) {
-        errors.push(
-          'Cannot specify an array of multiple point values for a question if real-time grading is disabled',
-        );
-      }
+      const effectiveAlternativeGroupAllowRealTimeGrading =
+        zoneQuestion.allowRealTimeGrading ?? zone.allowRealTimeGrading ?? allowRealTimeGrading;
+
       // We'll normalize either single questions or alternative groups
       // to make validation easier
-      let alternatives: QuestionPointsJson[] = [];
+      let alternatives: (QuestionPointsJson & { allowRealTimeGrading: boolean })[] = [];
```

Comment:
👀 What's up with this? Shouldn't you instead export/use a schema for QuestionAlternatives that is this union?

Reply:
Look at how the code is actually used. This is _not_ a reflection of any specific thing that exists in a JSON file/schema, it's just a temporary data structure that we use only for validation.